src/WebJobs.Script/OutOfProc/Http/DefaultHttpInvokerService.cs

```diff
@@ -3,23 +3,90 @@
 
 using System;
 using System.Net.Http;
+using System.Net.Http.Formatting;
 using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
 using Microsoft.Azure.WebJobs.Script.Description;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
 
-namespace Microsoft.Azure.WebJobs.Script.OutOfProc
+namespace Microsoft.Azure.WebJobs.Script.OutOfProc.Http
 {
     public class DefaultHttpInvokerService : IHttpInvokerService
     {
         private readonly HttpClient _httpClient;
+        private readonly string _httpInvokerBaseUrl;
 
         public DefaultHttpInvokerService(HttpClient httpClient)
         {
             _httpClient = httpClient;
+            // TODO: user agent ok?
+            _httpClient.DefaultRequestHeaders.UserAgent.ParseAdd(HttpInvokerConstants.UserAgentHeaderValue);
+            _httpInvokerBaseUrl = "http://localhost:8090/";
         }
 
-        public Task GetInvocationResponse(ScriptInvocationContext scriptInvocationContext)
+        public async Task InvokeAsync(ScriptInvocationContext scriptInvocationContext)
         {
-            throw new NotImplementedException();
+            var requestUri = _httpInvokerBaseUrl + scriptInvocationContext.FunctionMetadata.Name;
+            HttpScriptInvocationContext httpScriptInvocationContext = new HttpScriptInvocationContext();
+
+            // populate metadata
+            foreach (var bindingDataPair in scriptInvocationContext.BindingData)
+            {
+                if (bindingDataPair.Value != null)
+                {
+                    if (bindingDataPair.Value is HttpRequest)
+                    {
+                        // no metadata for httpTrigger
+                        continue;
+                    }
+                    // TODO: how to handle byte[]? base64 encoded string?
```

Comment:
Not sure if this is possible (or easy), but it would be cleaner if we actually did not support dataType:"binary" right now, but we allow users to specify dataType:"base64". Then we can make it a future enhancement to come up with a more efficient implementation for dataType:"binary". Thoughts?

Reply:
Updating ` dataType:base64` would require webjobs sdk changes. For now, we should pick a default on how we would send byte[] if we decide to support blob trigger as by default blob trigger data would be byte[]. 
Another option is to send HttpRequest as stream for binary data - out of scope for preview?

Reply:
I am OK with shipping the initial preview of this functionality with byte[] being converted to base64. Maybe you can file an issue for us to revisit (and maybe add base64 to webjobs sdk) later?

Reply:
opened https://github.com/Azure/azure-functions-host/issues/5016 thanks!